### ISSUE TITLE
[Observation] Running same image through Adain

### ISSUE THREAD

**Issue by ArturoDeza** [xunhuang1995/AdaIN-style#12]
I'm tweaking the code to do somewhat of a trivial example: Essentially running any image (in this case a scene, and not a texture), through the AdaIN, and having the style and content be the same image, with the goal of getting an exact reconstruction of the input image (as other style transfer methods can do). However, AdaIN seems to still texturize the output in this procedure, and the output looks a lot like an acuarela-like version of the same image, with most of the fine detail lost. I will try to post an example of this soon.

What is a good suggestion to work around this? I know the training algorithm is not out, but perhaps training the network with non-textures would improve such performance, or just more rounds of training? Thoughts?

**Comment by dovanchan** [comment]
content:
![image](https://cloud.githubusercontent.com/assets/26133721/24893819/99dade4c-1eb8-11e7-820f-90ca9b05c4b9.png)
style:
![image](https://cloud.githubusercontent.com/assets/26133721/24893868/f4ab91f4-1eb8-11e7-8ade-cfd0b41fad91.png)


I think the AdaIN need a lot of space for improvement,If it breakthrough,that will amazing

**Comment by gsssrao** [comment]
@ArturoDeza Could you post an example of input image used? 
Adaptive Instance Normalization should not cause any change. If you substitute x=y in equation (8) [here](https://arxiv.org/pdf/1703.06868.pdf) the output you get AdaIN(x,y)=x.
The only other reason for change can be that the decoder is not the exact inverse of encoder. This can be solved by training till Lc (refer Figure 2 of paper) is very low.

I tried to use `cornell.jpg` as both my input and style image. The output I got seems reasonable (I used `models/decoder-content-similar.t7`):
<p align='center'>
  <img src='https://cloud.githubusercontent.com/assets/9698886/24904239/79aa2280-1ecd-11e7-99f3-047ddf7bbde8.jpg' width="300px">
  <img src='https://cloud.githubusercontent.com/assets/9698886/24904252/85a99c28-1ecd-11e7-82d1-9807a813a2be.jpg' width="300px">
</p>

**Comment by dovanchan** [comment]
I think the result similar as fast-neural-style

**Comment by ArturoDeza** [comment]
this is what I get when I run th test.lua -content 3.png -style 3.png with the default values. Can someone please replicate this to make sure they get the same results?

![3](https://cloud.githubusercontent.com/assets/1279233/24919181/a89e28e2-1e97-11e7-857c-47dd0f0cd490.png)
![3_stylized_3](https://cloud.githubusercontent.com/assets/1279233/24919187/ae74ac28-1e97-11e7-8c89-5e8d52aec8eb.jpg)

My output when running th test.lua -style input/content/cornell.jpg -content input/content/cornell.jpg
![cornell_stylized_cornell](https://cloud.githubusercontent.com/assets/1279233/24919383/53f31f18-1e98-11e7-9cb4-c343570b95cb.jpg)
Looks a bit off, slightly less quality than yours @dovanchan

**Comment by gsssrao** [comment]
@ArturoDeza I too get similar results. It seems you are using the default decoder. Try the newer one. It has slightly better decoder weights.

`th test.lua -style input/content/cornell.jpg -content input/content/cornell.jpg -decoder models/decoder-content-similar.t7 `

With this you would get same results as mine. For the image you provided these are the results I get for the two decoders:

<p align='center'>
  <img src='https://cloud.githubusercontent.com/assets/9698886/24930320/bca37e3e-1f26-11e7-8027-36d6860c50e2.jpg' width="300px">
  <img src='https://cloud.githubusercontent.com/assets/9698886/24930361/d8d4335a-1f26-11e7-8db0-a4a53e64f9e8.jpg' width="300px">
</p>

As I had mentioned earlier, I think that the trained decoder is not the exact inverse of encoder. This can be solved by training till Lc (refer Figure 2 of paper) is very low i.e have more number of iterations.

**Comment by ArturoDeza** [comment]
Makes sense! Thanks, it gave me better results! Hoping the training code comes out soon.

**Comment by dovanchan** [comment]
@ArturoDeza  How about the result using my content and style image？Can you show to me？

**Comment by hristorv** [comment]
There is a port in Tensorflow on which i am working currently and we have  an issuem the pictures all appear to be darker and the color is a bit off. 
![paper](https://cloud.githubusercontent.com/assets/4991283/25035526/d28275e4-20f6-11e7-9a16-f3f67cab1925.png)

I would really appreciate some help take a look at the code here:

**Comment by ArturoDeza** [comment]
@dovanchan , Hmm, I still seem to get the same tile-like artifacts you are getting. I think this can be avoided with the training procedure that they use in the Diverse Synthesis paper. Attaching my output. *On the flipside -- notice that the style image you are inputting also has this heavy brush-like painting style which I think the AdaIN is capturing
![cat1_stylized_style1](https://cloud.githubusercontent.com/assets/1279233/25054241/d9adc8e0-2110-11e7-98d9-c22e14f58682.jpg)

**Comment by LalitPradhan** [comment]
@hristorv 
I'm facing the same issue while working with lasagne. Almost all pictures seem to have a blue colour. Did you get a solution to that?

**Comment by hristorv** [comment]
@LalitPradhan  I have solved the problem. The color values of the images are represented from 0 to 1. However they should be from 0 to 255 after postprocess.
What we need to do is preprocess the images, dividing by 255. And also postprocess, multiplying by 255 and clipping the values.
I am getting the same results as in the original paper and code.

**Comment by dovanchan** [comment]
@hristorv Can you use my content image and style image to have a test？（The cat photo I used before）

**Comment by LalitPradhan** [comment]
@hristorv 
Thanks. Your solution worked.

**Comment by MonaTanggg** [comment]
@ArturoDeza did you fix the lose problem yet, would you mind share your solution with lua ?

**Comment by ArturoDeza** [comment]
@MonaTanggg See this thread: https://github.com/xunhuang1995/AdaIN-style/issues/16

Essentially I trained a pix2pix Super-Resolution module that maps back to the original image. It does quite a good job removing the artifacts.

**Comment by akhauriyash** [comment]
Hello. I have been implementing this in tensorflow with reference to [this](https://github.com/jonrei/tf-AdaIN) . 
I am getting results with *VERY* low contrast, as in the pictures are extremely dull. Implementing the fast style transfer using a generator network gave much better results. 

Can someone tell me how I can correct this? Am i doing something wrong? I am also ensuring that the output image is converted to the 0 to 255 range. 

Thanks!
